packages/shared/src/lib/auth.ts

```diff
@@ -189,3 +193,22 @@ export const getNodeValue = (
   nodes: InitializationNode[],
 ): string =>
   nodes?.find(({ attributes }) => attributes.name === key)?.attributes?.value;
+
+export const isNativeAuthSupported = (provider: string): boolean =>
+  globalThis.webkit?.messageHandlers?.['native-auth'] &&
+  ['apple', 'google'].includes(provider);
+
+export const iosNativeAuth = async (
+  provider: string,
+): Promise<{ provider: string; token: string; nonce: string } | undefined> => {
+  globalThis.webkit.messageHandlers['native-auth'].postMessage(provider);
+  return new Promise((resolve) => {
+    globalThis.addEventListener(
```

Comment:
Can leak listeners if error happens or never responds

Reply:
maybe introduce a timeout, or even user can come back from native auth we should clear it?

Reply:
I'll check what can be done

Reply:
Added abort controller. I didn't want to introduce timeout so there will be no funky edge cases